### ISSUE TITLE
Something about the config.json file.

### ISSUE THREAD

**Issue by PKMMonster** [haamoon/finding_common_object#1]
Hello,your work is very meaningful for me,but I am puzzled with this config.json file, I can't understand what the parameters in it mean.

**Comment by AmirooR** [comment]
Hi @PKMMonster,

Here, I will write the explanation for [this](https://github.com/haamoon/finding_common_object/blob/master/experiments/mini/bs5/k8n20/config.json) test config file:
```python
{
  "test": {
    "k_shot": 8, # number of shots or positive bags (equivalent to number of nodes in graph)
    "bag_size": 5, # number of elements(images) in a bag (equivalent to number of labels in graph)
    "negative_bag_size": 20, # number of negative elements (should be multiple of bag_size in this implementation)
    "split": "test", # the split we want to sample problems from
    "num_sample_classes": 15, # maximum number of classes we want to sample images from for each problem
    "num_sample_classes_min": 5, # minimum number of classes we want to sample images from for each problem
    "shuffle": false, #whether to shuffle the iterators or not (should be true for training mode only)
    "dataset_name": "miniimagenet", # name of the dataset
    "use_features": true, # this should be always True in this implementation (uses the provided \
                          # pretrained features instead of images)
    "one_example_per_class": false, # if True, the elements in a bag are sampled in a way \
                          # that not any pair of them are from the same class. Otherwise, there might be \
                          # multiple elements in a bag that are from the same class. \
                          # this feature is only used for the few-shot classification experiment.
    "has_single_target": false, # if True, it ensures that only a single element in the bag is \
                          # sampled from the target class. This is also used for the few-shot experiment
    "add_gt_list": false, # this should always be False (it was used for visualization)
    "is_training": false, # this should be False in test and eval mode and True during training
    "prefetch_buffer_size": 50, # prefetch buffer size for tensorflow dataset's queue
    "iters": 1000, # total number of random problems to sample from
    "checkpoint": "experiments/mini/k2n20/train", # this is a checkpoint folder for the trained pairwise similarity function. \
                           # used to compute pairwise potentials in the graphical model.
    "unary_scales": [0.3, 0.6, 1.2, 1.9], # These are the scalar values multiplied to unary scores at each level of our greedy \
                           # tree(from bottom to top level) and are found by cross validation.
    "ntop_proposals": 300, # number of top common object proposals selected after pruning at each level of the greedy method
    "print_freq": 100, # print some info after every 100 sampled problems(iterations).
    "regularization_scale": 5e-4, # this is not important during testing but I guess is passed to the model constructor
    "seed": 1357 # random seed
  }
}
```

Please let me know if anything is missing.

Best,
Amir

**Comment by PKMMonster** [comment]
Thank you so much! The "lr_schedule" in the training parameters also confused me. Even though I knew what they meant, but I didn't know how they were obtained. Or I'm wrong about "boundaries" and "values".
Thank you again for your help！

**Comment by AmirooR** [comment]
@PKMMonster , the lr_schedule is used in [utils.py](https://github.com/haamoon/finding_common_object/blob/master/utils.py#L11). This is how piecewise constant learning rate schedule is used in `tf.keras.optimizer` e.g., see [here](https://www.tensorflow.org/api_docs/python/tf/keras/optimizers/schedules/PiecewiseConstantDecay).

**Comment by PKMMonster** [comment]
Ok,I got it! Thank you for your kind reply.
